examples/advanced/processes.ipynb

```diff
@@ -0,0 +1,571 @@
+      "    a_voltage = nengo.Probe(a.neurons, 'voltage')\n",
+      "    b_voltage = nengo.Probe(b.neurons, 'voltage')\n",
+      "    \n",
+      "with nengo.Simulator(model) as sim:\n",
+      "    sim.run(0.15)\n",
+      "    \n",
+      "plt.plot(sim.trange(), sim.data[a_voltage], label=\"deterministic\")\n",
+      "plt.plot(sim.trange(), sim.data[b_voltage], label=\"noisy\")\n",
+      "plt.xlabel('time [s]')\n",
+      "plt.ylabel('voltage')\n",
+      "plt.legend(loc=4);"
+     ],
+     "language": "python",
+     "metadata": {},
+     "outputs": []
+    },
+    {
+     "cell_type": "markdown",
+     "metadata": {},
+     "source": [
+      "We see that the neuron without noise (blue)\n",
+      "approaches its firing threshold,\n",
+      "but never quite gets there.\n",
+      "Adding a bit of noise (green)\n",
+      "causes the neuron to occasionally jitter above the threshold,\n",
+      "resulting in two spikes\n",
+      "(where the voltage suddenly drops to zero)."
+     ]
+    },
+    {
+     "cell_type": "markdown",
+     "metadata": {},
+     "source": [
+      "## `FilteredNoise`\n",
+      "\n",
+      "The `FilteredNoise` process takes a white noise signal\n",
+      "and passes it through a filter.\n",
+      "Using any type of lowpass filter (e.g. `Lowpass`, `Alpha`)\n",
+      "will result in a signal similar to `WhiteSignal`,\n",
+      "but rather than being ideally filtered\n",
+      "(i.e. no frequency content above the cutoff),\n",
+      "the `FilteredNoise` signal\n",
+      "will have some frequency content above the cutoff,\n",
+      "with the amount depending on the filter used.\n",
+      "Here, we can see how an `Alpha` filter\n",
+      "(a second-order lowpass filter)\n",
+      "is much better than the `Lowpass` filter\n",
+      "(a first-order lowpass filter)\n",
+      "at removing the high-frequency content."
+     ]
+    },
+    {
+     "cell_type": "code",
+     "collapsed": false,
+     "input": [
+      "process1 = nengo.processes.FilteredNoise(\n",
+      "    dist=nengo.dists.Gaussian(0, 0.01), \n",
+      "    synapse=nengo.Alpha(0.005),\n",
+      "    seed=0)\n",
+      "\n",
+      "process2 = nengo.processes.FilteredNoise(\n",
+      "    dist=nengo.dists.Gaussian(0, 0.01), \n",
+      "    synapse=nengo.Lowpass(0.005),\n",
+      "    seed=0)\n",
+      "\n",
+      "tlen = 0.5\n",
+      "plt.plot(process1.trange(tlen), process1.run(tlen))\n",
+      "plt.plot(process2.trange(tlen), process2.run(tlen));"
+     ],
+     "language": "python",
+     "metadata": {},
+     "outputs": []
+    },
+    {
+     "cell_type": "markdown",
+     "metadata": {},
+     "source": [
+      "The `FilteredNoise` process with an `Alpha` synapse (blue)\n",
+      "has significantly lower high-frequency components\n",
+      "than a similar process with a `Lowpass` synapse (green)."
+     ]
+    },
+    {
+     "cell_type": "markdown",
+     "metadata": {},
+     "source": [
+      "## `PresentInput`\n",
```

Comment:
It would be great to have a process for `utils.functions.piecewise`. I'd keep it separate from `PresentInput`, though, or at least have some version of it with fixed times (maybe if you pass a constant it could have fixed times, or a list of times it could have varying?). It's just easier to do in OCL if the times are fixed.

Also with piecewise it would be great to have linear interpolation, too.